staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -1329,24 +1329,32 @@ func mergeMap(original, patch map[string]interface{}, schema LookupPatchMeta, me
 			}
 		}
 
-		_, ok := original[k]
-		if !ok {
-			// If it's not in the original document, just take the patch value.
-			original[k] = patchV
-			continue
-		}
+		var originalKind reflect.Kind
 
-		originalType := reflect.TypeOf(original[k])
-		patchType := reflect.TypeOf(patchV)
-		if originalType != patchType {
-			original[k] = patchV
-			continue
+		_, foundInOriginal := original[k]
+		if patchV != nil {
+			if !foundInOriginal {
+				original[k] = reflect.Indirect(reflect.New(reflect.TypeOf(patchV))).Interface()
```

Comment:
Can you add a comment to explain what do you want to do here?

Reply:
sure 😉